web/client/src/library/pages/ide/IDE.tsx

```diff
@@ -228,15 +237,30 @@ export default function PageIDE(): JSX.Element {
   }
 
   function restoreEditorTabsFromSaved(files: ModelFile[]): void {
-    files.forEach(file => {
-      if (storedTabsIds.includes(file.id)) {
-        const tab = createTab(file)
-
-        if (storedTabsId === file.id) {
-          selectTab(tab)
-        } else {
-          addTab(tab)
-        }
+    console.log('restoreEditorTabsFromSaved', {
+      storedTabsIds,
+      storedTabsId,
+    })
+    if (isArrayEmpty(storedTabsIds)) return
+
+    storedTabsIds.forEach(({ id, content }) => {
+      const file = files.find(file => file.id === id)
+      let storedTab: EditorTab
+
+      if (isNil(file)) {
+        storedTab = isNil(tab) || tab.file.isChanged ? createTab() : tab
+
+        storedTab.file.content = content ?? ''
+      } else {
+        storedTab = createTab(file)
+
+        storedTab.file.content = content ?? storedTab.file.content
+      }
+
+      addTab(storedTab)
+
+      if (storedTabsId === id || (isNil(id) && storedTabsId === content)) {
```

Comment:
So `storedTabsId` can equal id or content? Shouldn't it only ever be able to equal id?